### PR TITLE
Branch002

### PR DESCRIPTION
**Pull request by Poikka001**

### CODE REVIEW

File: src/main/java/mate/academy/intro/model/Category.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Column(name = "description")
+    private String description;
+
+    @Column(nullable = false, name = "is_deleted")
+    private boolean isDeleted = false;
+}
+
+
+
```
Comment (kerrrusha) [inline src/main/java/mate/academy/intro/model/Category.java:40]: you need exactly 1 blank line at the EOF

File: src/main/java/mate/academy/intro/repository/Category/CategoryRepository.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package mate.academy.intro.repository.Category;
+
+import mate.academy.intro.model.Category;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+
+public interface CategoryRepository extends JpaRepository<Category,
+        Long>, JpaSpecificationExecutor<Category> {
+}
```
Comment (kerrrusha) [inline src/main/java/mate/academy/intro/repository/Category/CategoryRepository.java:9]: you need exactly 1 blank line at the EOF

File: src/main/java/mate/academy/intro/repository/book/BookRepository.java [hunk 3]
```diff
@@ -4,7 +4,13 @@
...
-public interface BookRepository extends JpaRepository<Book,
-        Long>, JpaSpecificationExecutor<Book> {
-}
+import java.util.List;
 
+//public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+//    List<Book> findAllByCategoryId(Long categoryId);
+//}
+
+
```
Comment (kerrrusha) [inline src/main/java/mate/academy/intro/repository/book/BookRepository.java:13]: such trash should be deleted

File: src/main/java/mate/academy/intro/repository/book/BookRepository.java [hunk 4]
```diff
@@ -4,7 +4,13 @@
...
+public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book>  {
+    List<Book> findAllByCategoriesId(Long categoryId);
+}
```
Comment (kerrrusha) [inline src/main/java/mate/academy/intro/repository/book/BookRepository.java:16]: you need exactly 1 blank line at the EOF

File: src/main/java/mate/academy/intro/service/BookServiceImpl.java [hunk 5]
```diff
@@ -54,6 +66,6 @@ public void deleteBook(Long bookId) {
     @Override
     public Page<BookDto> search(Pageable pageable, Map<String, String[]> params) {
         return bookRepository.findAll(bookSpecificationBuilder.build(params), pageable)
-                .map(bookMapper::toDto);
+                .map(bookMapper::toDtoBook);
     }
 }
```
Comment (kerrrusha) [inline src/main/java/mate/academy/intro/service/BookServiceImpl.java:71]: you need exactly 1 blank line at the EOF

File: src/main/java/mate/academy/intro/mapper/BookMapper.java [hunk 6]
```diff
@@ -2,15 +2,19 @@
...
+    BookDto toDtoBook(Book model);
+
     @Mapping(target = "id", ignore = true)
     @Mapping(target = "deleted", ignore = true)
-    Book toModel(CreateBookRequestDto dto);
-}
+    Book toEntity(CreateBookRequestDto bookDto);
+
+    BookDtoWithoutCategoryIds toDtoWithoutCategories (Book book);
+}
```
Comment (kerrrusha) [inline src/main/java/mate/academy/intro/mapper/BookMapper.java:20]: you need exactly 1 blank line at the EOF

File: src/main/java/mate/academy/intro/controller/CategoryController.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        if (auth.getName().startsWith("admin")) categoryService.deleteCategory(categoryId);
+    }
+
+    @GetMapping("/{bookId}/books")
+    @Operation(description = "Get books by category ID")
+    @ApiResponse(responseCode = "200", description = "Books by category ID")
+    public List<BookDto> findAllByCategoryId(@PathVariable Long bookId) {
+        return bookService.findAllByCategoryId(bookId);
+    }
+}
```
Comment (kerrrusha) [inline src/main/java/mate/academy/intro/controller/CategoryController.java:86]: you need exactly 1 blank line at the EOF

File: src/main/java/mate/academy/intro/controller/BookController.java [hunk 8]
```diff
@@ -75,6 +70,7 @@ public BookDto updateBook(@PathVariable Long bookId,
     @Operation(description = "Delete book by Id")
     @ApiResponse(responseCode = "200", description = "Delete book")
     public void deleteBook(@PathVariable Long bookId) {
-        bookService.deleteBook(bookId);
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        if (auth.getName().startsWith("admin")) bookService.deleteBook(bookId);
     }
 }
```
Comment (kerrrusha) [inline src/main/java/mate/academy/intro/controller/BookController.java:76]: you need exactly 1 blank line at the EOF

**Review by kerrrusha — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/intro/repository/book/BookRepository.java (hunk 3 above)
Comment (Poikka001) [reply src/main/java/mate/academy/intro/repository/book/BookRepository.java:13]: my mistake)

**Review by kerrrusha — APPROVED** [review]

**Pull request merged**